src/mapreduce.jl

```diff
@@ -1,25 +1,24 @@
-# Returns the common Size of the inputs (or else throws a DimensionMismatch)
-@inline same_size(a1::StaticArray, as::StaticArray...) = _same_size(Size(a1), as...)
-@inline _same_size(s::Size) = s
-@inline function _same_size(s::Size, a1::StaticArray, as::StaticArray...)
-    if s === Size(a1)
-        return _same_size(s, as...)
-    else
-        throw(DimensionMismatch("Dimensions must match. Got inputs with $s and $(Size(a1))."))
-    end
-end
-
 @inline _first(a1, as...) = a1
 
 ################
 ## map / map! ##
 ################
 
-@inline function map(f, a::StaticArray, b::StaticArray...)
-    _map(f, same_size(a, b...), a, b...)
-end
-
-@generated function _map(f, ::Size{S}, a::StaticArray...) where {S}
+@inline map(f, as::StaticArray...) =
+    _map(f, same_size(as...), as...)
+# Mixed StaticArray + AbstractArray; various versions to avoid ambiguities.
+# With the versions below, if a StaticArray isn't present in the first two
+# arguments, we'll end up in Base.map() instead.
+@inline map(f, a1::StaticArray, as::AbstractArray...) =
```

Comment:
Yesterday, I saw an unofficial "trick" to get "at least one of these" that goes something like:
```julia
map(Union{SA, AbstractArray}...) where {SA <: StaticArray} = ...
```
It works because the method is ignored unless `SA` can be found and determined (all method parameters must be specified in order to be counted as a match). Not sure if it will work for us (what if there are multiple `SA` types) and not sure if it's a good idea (it could be fragile, also Keno or someone was saying the requirement for `SA` to be determinable might be dropped - but I didn't see how the method body could make sense in this case).